src/wallet/mod.rs

```diff
@@ -454,6 +464,170 @@ impl Wallet {
         Ok(())
     }
 
+    fn create_deposit_op_drivechain_output(
+        sidechain_number: SidechainNumber,
+        sidechain_ctip_amount: Amount,
+        value_sats: Amount,
+    ) -> bdk::bitcoin::TxOut {
+        let deposit_txout =
+            messages::create_m5_deposit_output(sidechain_number, sidechain_ctip_amount, value_sats);
+
+        bdk::bitcoin::TxOut {
+            script_pubkey: bdk::bitcoin::ScriptBuf::from_bytes(
+                deposit_txout.script_pubkey.to_bytes(),
+            ),
+            value: deposit_txout.value.to_sat(),
+        }
+    }
+
+    async fn fetch_ctip_transaction(&self, ctip_txid: Txid) -> Result<bdk::bitcoin::Transaction> {
+        let block_hash = None;
+
+        let rpc_res = self
+            .main_client
+            .get_raw_transaction(ctip_txid, GetRawTransactionVerbose::<true>, block_hash)
+            .await
+            .into_diagnostic()?;
+
+        let Some(transaction_hex) = rpc_res.as_str() else {
+            return Err(miette!("failed to fetch ctip transaction"));
+        };
+
+        let transaction_bytes = hex::decode(transaction_hex).unwrap();
+        let mut cursor = Cursor::new(transaction_bytes);
```

Comment:
Instead of using `Cursor`, you can use `https://docs.rs/bitcoin/latest/bitcoin/consensus/encode/fn.deserialize_hex.html` or `https://docs.rs/bitcoin/latest/bitcoin/consensus/fn.deserialize.html`